Support multiple homes / servers

TAKeanice opened the pull request:
As requested in #557, in this implementation, I pieced together a simple yet effective way to manage multiple homes in OpenHab ios.
The implementation consists largely of two parts:

1. New preferences: 
    - stored preferences are sets of preferences that were stored, each corresponding to one server / home
    - "homeName" to associate names with the homes represented by stored preferences
    - currently used settings to identify which of the stored preferences are in use right now
2. A view to choose and manage sets of preferences (i.e. homes / servers) accessible from the right side menu

The code is quite rough at this point, but everything including the upgrade (and downgrade) of the app should work smoothly. I am opening this PR early to gather your feedback.

TAKeanice committed:
enable storing multiple sets of preferences

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
add option for home selection to drawer menu

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
view to switch between homes started

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
log preference values and value changes, store preferences correctly

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
prevent concurrent access by read and write of settings

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
controls for editing homes list, renaming and deleting entries

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
implement home removal and renaming

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
improve homeselection home management

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
init state for home selection view correctly

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice committed:
make whole cell selectable in home selection

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

timbms commented:
Awesome @TAKeanice 
I am currently reworking the app in the branch `openapigen`. Actually this has evolved into much more than the adoption of Apple's OpenAPI generator. It also includes a rework of the Settings interface rebuild in SwiftUI:
[IMG_2163.pdf](https://github.com/user-attachments/files/20556873/IMG_2163.pdf)
It would make a lot of sense, if you could port your proposal to openapigen.

TAKeanice committed:
remove localizations to satisfy localization tests

this sounds paradoxical, but the localization test does not accept a localization that does not exist in _all_ languages at once.

Signed-off-by: Tassilo Karge <tassilo.karge@web.de>

TAKeanice commented:
> It also includes a rework of the Settings interface rebuild in SwiftUI: [IMG_2163.pdf](https://github.com/user-attachments/files/20556873/IMG_2163.pdf) It would make a lot of sense, if you could port your proposal to openapigen.

I will try and port it to your branch, but the settings UI remains untouched by my changes.
